File: client/src/models/Restaurant.js
import { DatabaseItem } from "./DatabaseItem.js";
import { Profile } from "./Profile.js";

export class Restaurant extends DatabaseItem {
  constructor(data) {
    super(data)
    this.name = data.name
    this.imgUrl = data.imgUrl
    this.description = data.description
    this.visits = data.visits
    this.isShutdown = data.isShutdown
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
    this.reportCount = data.reportCount
  }
}

File: server/Models/Restaurant.cs
namespace help_reviews.Models;

public class Restaurant : RepoItem<int>
{
  public string Name { get; set; }
  public string ImgUrl { get; set; }
  public string Description { get; set; }
  public int Visits { get; set; }
  public bool? IsShutdown { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public int ReportCount { get; set; }
}
